rasa/nlu/components.py

```diff
@@ -87,13 +87,40 @@ def validate_required_components_from_data(
     pipeline: List["Component"], data: TrainingData
 ):
 
-    response_selector_exists = False
-    for component in pipeline:
-        # check if a response selector is part of NLU pipeline
-        if MESSAGE_RESPONSE_ATTRIBUTE in component.provides:
-            response_selector_exists = True
+    # check if there are regex features
+    if data.regex_features and not any(
+        [component.name == "RegexFeaturizer" for component in pipeline]
+    ):
+        warnings.warn(
+            "You have defined training data with Regex's, but "
+            "haven't setup RegexFeaturizer in your config. "
```

Comment:
```suggestion
            "your NLU pipeline does not include a RegexFeaturizer. "
```

Reply:
(reflect these changes in the other ones)

Reply:
cleaned these up accordingly